docs/DeveloperGuide.md

```diff
@@ -1,38 +1,703 @@
+7. `Validation.splitAndValidateGymStationInput` is called to both split and validate the user's input for the gym station.
+    - This method calls `Validation.validateWeightsArray` to ensure the `weights` input is valid.
+    - `CustomExceptions.InsufficientInput` is thrown if either not enough parameters are specified or blank parameters are found.
+    - `CustomExceptions.InvalidInput` is thrown if the parameters passed in are invalid and do not follow the stipulated format.
+
+8. If valid, parameters are converted into either `int` for the number of repetitions and sets, or `Arraylist<Integer>` for the weights array. The parameters are then passed to `gym.addStation` to add a `GymStation` object to the new `Gym` object.
+
+9. `Output.printAddGym()` is then called to print the message acknowledging the successful adding of a new `Gym` object.
+
+###### [Back to table of contents](#table-of-contents)
+
+--- 
+
+### Health
+
+#### Add Period
+<code style="color: #D85D43;">
+HEALTH /h:period /start:[start_date] /end:[end_date]
+</code>
+
+- `[start_date]`  and `[end_date]` are in DD-MM-YYYY format.
+
+##### Period Sequence
+
+The sequence diagram below illustrates the process of period prediction.
+
+1. The Handler class receives `userInput` through `Handler.processInput()` which calls `Handler.handleHealth` to handle health-related operations.
+
+2. `Parser.extractSubstringaFromSpecificIndex()` is then called to extract `typeOfHealth`, in this case, `prediction`.
+
+3. The Handler class then calls `Parser.parsePredictionInput()`.
+
+4. `HealthList.getPeriodSize()` is then called to retrieve the size of `Periods` to ensure sufficient `Period` inputs are present for prediction.
+
+5. `HealthList.printLatestThreeCycles()` prints the user's latest three menstrual cycles.
+
+6. `HealthList.predictNextPeriodStartDate()` gets the predicted start date by calling `Period.nextCyclePrediction()`.
+
+7. `Period.getLastThreeCycleLengths()` is called to find the sum of the latest three cycle lengths before returning to `Period.nextCyclePrediction()` for necessary calculation to obtain the predicted start date.
+
+8. The `Parser` class calls `Period.printNextCyclePrediction()` with the predicted start date as a parameter. This method prints a string indicating the number of days until the predicted start date of the next period, or how many days late the period is if the current date is after the predicted start date.
+
+![Period Sequence Diagram](img/period_sequence.png)
+
+![Period Validation Diagram](img/period_validation.png)
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Add BMI
+
+<code style="color: #D85D43;">
+HEALTH /h:bmi /height:[height] /weight:[weight] /date:[date]
+</code>
+
+- `[height]` is a 2 **decimal place positive number** representing the user's height.
+
+- `[weight]`is a 2 **decimal place positive number** representing the user's weight.
+
+- `[date]` is in `DD-MM-YYYY` format (i.e. `19-03-2024`).
```

Comment:
height, weight and date all have specific inputs that have been defined "decimal place positive number" which is good practice!